FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/PrivateKeysViewModel.kt

```diff
@@ -597,41 +519,64 @@ class PrivateKeysViewModel(application: Application) : AccountViewModel(applicat
    */
   private suspend fun registerUserPublicKey(
     accountEntity: AccountEntity,
-    keyDetails: PgpKeyDetails
+    keyDetails: PgpKeyDetails,
+    idToken: String? = null,
+    isSilent: Boolean = true,
   ): Boolean = withContext(Dispatchers.IO) {
-    return@withContext try {
-      val model = InitialLegacySubmitModel(accountEntity.email, keyDetails.publicKey)
-      val initialLegacySubmitResult = apiRepository.postInitialLegacySubmit(getApplication(), model)
-      when (initialLegacySubmitResult.status) {
-        Result.Status.SUCCESS -> {
-          val body = initialLegacySubmitResult.data
-          body != null && (body.apiError == null || body.apiError.code !in 400..499)
+    val submitPubKeyResult = apiRepository.submitPubKey(
+      context = getApplication(),
+      email = accountEntity.email,
+      pubKey = keyDetails.publicKey,
+      idToken = idToken,
+      orgRules = accountEntity.clientConfiguration
+    )
+
+    when (submitPubKeyResult.status) {
+      Result.Status.SUCCESS -> {
+        val body = submitPubKeyResult.data
+        val isSuccess = body != null && (body.apiError == null || body.apiError.code !in 400..499)
+        if (isSilent) {
+          isSuccess
+        } else {
+          throw IllegalStateException(body?.apiError?.msg)
```

Comment:
1. `registerUserPublicKey` function parameters (idToken, isSilent) defintion(comment) is missing
2. If `isSilent` is false, then we throw error if though `isSuccess` is true?

Reply:
> If isSilent is false, then we throw error if though isSuccess is true?

Good one. My bad. I've fixed it.